lib/services/http-service.js

```diff
@@ -190,90 +181,113 @@ function httpServiceFactory(
-                console.log("HTTP server closing");
-            });
-
-            promises.push(new Promise(function (resolve, reject) {
-                httpServer.listen(
-                    configuration.get('httpBindPort', 80),
-                    configuration.get('httpBindAddress', '0.0.0.0'),
-                    function (error) { return error ? reject(error) : resolve(); }
-                );
-            }));
-
-            servers.push(httpServer);
-        }
-
-        if (configuration.get('httpsEnabled', false)) {
-            var httpsOptions;
-            var pfx = configuration.get('httpsPfx', null);
+        _.forEach(endpoint.routers, function(routerName) {
+            var router;
+            try {
+                router = injector.get(routerName);
+            } catch (e){
+                logger.error('Cound not found router named : ' + routerName, { error: e });
+                return;
+            }
+            //router.authEnabled = endpoint.authEnabled;
+            router.endpoint = endpoint;
+            router.mount();
+            app.use(/\/api\/(common|current|1\.1)/, router);
+        });
 
-            if (pfx) {
-                httpsOptions = {
-                    pfx: fs.readFileSync(pfx)
+        if (endpoint.httpsEnabled) {
+            if (endpoint.httpsPfx) {
+                this.httpsOptions = {
+                    pfx: fs.readFileSync(endpoint.httpsPfx)
                 };
             } else {
-                httpsOptions = {
-                    cert: fs.readFileSync(
-                        configuration.get('httpsCert', './data/dev-cert.pem')
-                    ),
-                    key: fs.readFileSync(
-                        configuration.get('httpsKey', './data/dev-key.pem')
-                    )
+                this.httpsOptions = {
+                    cert: fs.readFileSync(endpoint.httpsCert),
+                    key: fs.readFileSync(endpoint.httpsKey)
                 };
             }
+        }
+    };
 
-            var httpsServer = https.createServer(httpsOptions, this);
+    HttpService.prototype._parseEndpoint = function(endpoint) {
+        function parseRouterNames (routers) {
+            if (_.isEmpty(routers)) {
+                return ['northbound-api-router', 'southbound-api-router'];
+            } else {
+                if (_.isString(routers)) {
+                    return [routers];
+                }
+                if (_.isArray(routers) && _.all(routers, _.isString)) {
+                    return routers;
+                }
+                return ['northbound-api-router', 'southbound-api-router'];
+            }
+        }
 
-            wss.start(httpsServer, {
-                secureOptions: httpsOptions
-            });
+        return {
+            address: endpoint.address || '0.0.0.0',
+            port: endpoint.port || (endpoint.httpsEnabled ? 443 : 80),
+            httpsEnabled : endpoint.httpsEnabled === true,
```

Comment:
Yes, that will also work. But think about illegal input `httpsEnabled: 'Off'` in config file.